### PR TITLE
fix escape nested comments in C++

### LINKED ISSUE #115243: [cppdocs] torch::load function istringstream example typo
### 📚 The doc issue

Currently the docs show the following:
```c++
torch::nn::Linear model(3, 4);
torch::load(model, "model.pt");

torch::optim::SGD sgd(/*lr=*&zwj;/0.9);
std::istringstream stream("...");
torch::load(sgd, stream);

auto tensor = torch::ones({3, 4});
torch::load(tensor, "my_tensor.pt");
```

The comments that start with `sgd(/*` is never closed. This is probably not intentional.

### Suggest a potential alternative/fix

The following snippet can be used instead
```c++
torch::nn::Linear model(3, 4);
torch::load(model, "model.pt");

torch::optim::SGD sgd(/*lr=*/0.9);
std::istringstream stream("...");
torch::load(sgd, stream);

auto tensor = torch::ones({3, 4});
torch::load(tensor, "my_tensor.pt");
```

cc @svekars @carljparker @jbschlosser

### PR DESCRIPTION
**Pull request by lancerts**
Fixes #115243, as it is tricky to deal with the nested comment in doxygen + sphinx. Change 6 below is adopted as the fix. All other changes do not work.

After adopting change 6, realize the original
`torch::optim::SGD sgd(0.9);` is not the correct call to the sgd constructor,
modified to the correct one
`torch::optim::SGD sgd(model->parameters(), 0.9);`

cc @svekars @brycebortree @jbschlosser

- Original in [link](https://pytorch.org/cppdocs/api/function_namespacetorch_1ad98de93d4a74dd9a91161f64758f1a76.html#exhale-function-namespacetorch-1ad98de93d4a74dd9a91161f64758f1a76): `///   torch::optim::SGD sgd(/*lr=*/0.9);`
image

- Change 1, this solution is referenced from [here](https://stackoverflow.com/questions/24978463/doxygen-escape-nested-comments-in-c): `///   torch::optim::SGD sgd(/&zwj;* lr= *&zwj;/0.9);`
image

- Change 2: `///   torch::optim::SGD sgd(/* lr= *//* 0.9);`
image

- Change 3: `///   torch::optim::SGD sgd(/\*lr=\*/0.9);`
image

- Change 4: `///   torch::optim::SGD sgd(/&lowast; lr= &lowast;/0.9);`
image

- Change 5: 
```
/// \rst
/// .. code-block:: cpp
///
///   torch::nn::Linear model(3, 4);
///   torch::load(model, "model.pt");
///   \verbatim
///   torch::optim::SGD sgd(/*lr=*/0.9);
///   \endverbatim
///   std::istringstream stream("...");
///   torch::load(sgd, stream);
///
///   auto tensor = torch::ones({3, 4});
///   torch::load(tensor, "my_tensor.pt");
/// \endrst
```
image

- Change 6: `///   torch::optim::SGD sgd(0.9);  // 0.9 is the learning rate`
image
image

### CODE REVIEW

**Comment by cpuhrsch** [comment]
@lancerts - I approved running this PR since it fixes a confusing doc issue. I'll take a look later at the built docs and verify this fixed it. Thank you!

**Comment by lancerts** [comment]
> @lancerts - I approved running this PR since it fixes a confusing doc issue. I'll take a look later at the built docs and verify this fixed it. Thank you!

Cool, tested different fixes and attached the corresponding docs rendered in the above.

**Comment by cpuhrsch** [comment]
cc @janeyx99 FYI

**Review by cpuhrsch — APPROVED** [review]

**Comment by lancerts** [comment]
@malfet @cpuhrsch May I know what is the issue with mergeability? Thanks

**Comment by malfet** [comment]
@lancerts no idea, feel free to ignore

**Review by malfet — APPROVED** [review]
